tests/FrameCameraTests.cpp

```diff
@@ -73,31 +95,106 @@ TEST_F(FrameSensorModel, SlightlyOffCenter) {
    EXPECT_NEAR(groundPt.x, 0.44979759, 1e-8);
    EXPECT_NEAR(groundPt.y, -14.99325304, 1e-8);
    EXPECT_NEAR(groundPt.z, 14.99325304, 1e-8);
 }
 TEST_F(FrameSensorModel, OffBody2) {
-   csm::ImageCoord imagePt(-0.5, 14.5);
+   csm::ImageCoord imagePt(0.0, 15.0);
    csm::EcefCoord groundPt = sensorModel->imageToGround(imagePt, 0.0);
    EXPECT_NEAR(groundPt.x, 0.44979759, 1e-8);
    EXPECT_NEAR(groundPt.y, 14.99325304, 1e-8);
    EXPECT_NEAR(groundPt.z, -14.99325304, 1e-8);
 }
 TEST_F(FrameSensorModel, OffBody3) {
-   csm::ImageCoord imagePt(-0.5, -0.5);
+   csm::ImageCoord imagePt(0.0, 0.0);
    csm::EcefCoord groundPt = sensorModel->imageToGround(imagePt, 0.0);
    EXPECT_NEAR(groundPt.x, 0.44979759, 1e-8);
    EXPECT_NEAR(groundPt.y, 14.99325304, 1e-8);
    EXPECT_NEAR(groundPt.z, 14.99325304, 1e-8);
 }
 TEST_F(FrameSensorModel, OffBody4) {
-   csm::ImageCoord imagePt(14.5, 14.5);
+   csm::ImageCoord imagePt(15.0, 15.0);
    csm::EcefCoord groundPt = sensorModel->imageToGround(imagePt, 0.0);
    EXPECT_NEAR(groundPt.x, 0.44979759, 1e-8);
    EXPECT_NEAR(groundPt.y, -14.99325304, 1e-8);
    EXPECT_NEAR(groundPt.z, -14.99325304, 1e-8);
 }
 
+
+// Focal Length Tests:
+TEST_F(FrameIsdTest, FL500_OffBody4) {
+   isd.clearParams("focal_length");
+   isd.addParam("focal_length","500.0");
+   UsgsAstroFramePlugin frameCameraPlugin;
+         
+   csm::Model *model = frameCameraPlugin.constructModelFromISD(
+         isd,
+         "USGS_ASTRO_FRAME_SENSOR_MODEL");
+   
+   UsgsAstroFrameSensorModel* sensorModel = dynamic_cast<UsgsAstroFrameSensorModel *>(model);
+   
+   ASSERT_NE(sensorModel, nullptr);
+   csm::ImageCoord imagePt(15.0, 15.0);
+   csm::EcefCoord groundPt = sensorModel->imageToGround(imagePt, 0.0);
+   EXPECT_NEAR(groundPt.x, 9.77688917, 1e-8);
+   EXPECT_NEAR(groundPt.y, -1.48533467, 1e-8);
+   EXPECT_NEAR(groundPt.z, -1.48533467, 1e-8);
+}
+TEST_F(FrameIsdTest, FL500_OffBody3) {
+   isd.clearParams("focal_length");
+   isd.addParam("focal_length","500.0");
+   UsgsAstroFramePlugin frameCameraPlugin;
+         
+   csm::Model *model = frameCameraPlugin.constructModelFromISD(
+         isd,
+         "USGS_ASTRO_FRAME_SENSOR_MODEL");
+   
+   UsgsAstroFrameSensorModel* sensorModel = dynamic_cast<UsgsAstroFrameSensorModel *>(model);
+   
+   ASSERT_NE(sensorModel, nullptr);
+   csm::ImageCoord imagePt(0.0, 0.0);
+   csm::EcefCoord groundPt = sensorModel->imageToGround(imagePt, 0.0);
+   EXPECT_NEAR(groundPt.x, 9.77688917, 1e-8);
+   EXPECT_NEAR(groundPt.y, 1.48533467, 1e-8);
+   EXPECT_NEAR(groundPt.z, 1.48533467, 1e-8);
+}
+TEST_F(FrameIsdTest, FL500_Center) {
+   isd.clearParams("focal_length");
+   isd.addParam("focal_length","500.0");
+   UsgsAstroFramePlugin frameCameraPlugin;
+         
+   csm::Model *model = frameCameraPlugin.constructModelFromISD(
+         isd,
+         "USGS_ASTRO_FRAME_SENSOR_MODEL");
+   
+   UsgsAstroFrameSensorModel* sensorModel = dynamic_cast<UsgsAstroFrameSensorModel *>(model);
+   
```

Comment:
All these repeated instantiations look good. We can refactor out once parameterized tests are working the way we want them to.